src/parser/tokenizer/yaml/index.js

```diff
@@ -0,0 +1,33 @@
+const yaml = require('js-yaml');
+const { flatten, uniq } = require('lodash');
+
+const toArray = (obj, arr) => {
+  if (obj) {
+    Object.keys(obj).forEach((key) => {
+      const value = obj[key];
+      if (value !== null && typeof value === 'object') {
+        toArray(value, arr)
```

Comment:
As it's using recursion, should we ensure that there is a hard limit of i.e. 100 iterations, wdyt?

Reply:
for a nested object from a yaml file i think it's fine not to set a hard limit as it shouldn't get near node's max stack size, i can still implement it if you prefer though?

Reply:
In theory, recursion will consume more stack space than an iterative algorithm so it's better to have a circuit-breaker. Let's set a hard limit of 50 recursive calls. I doubt it there are yaml documents that have elements nested at this level, usually it's not more than 3 or 4.

Reply:
ok i've added the limit, let me know if that looks good